src/dtx/dtx_resync.c

```diff
@@ -518,7 +518,7 @@ dtx_iter_cb(uuid_t co_uuid, vos_iter_entry_t *ent, void *args)
 	} else {
 		/* For pool map refresh. */
 		/* Leader: do nothing. */
-		if ((ent->ie_dtx_flags & DTE_LEADER) && dra->resync_version != dra->discard_version)
+		if ((ent->ie_dtx_flags & DTE_LEADER) && dra->discard_version == 0)
```

Comment:
@Nasf-Fan should probably look at these changes since he's the one that changed that from a boolean.

Reply:
Both the current logic and the patch are not suitable. I have made another patch to fix related issue:

https://github.com/daos-stack/daos/pull/9825

Reply:
Hmm, this is enough for discard case, since discard_version is only set for discard case?  

Would you mind fix this base on this patch. 

Sorry, this is 2.2 blocker.

Reply:
> Hmm, this is enough for discard case, since discard_version is only set for discard case?

discard version will be set as long as the target status is ST_UP (in reintegration), that may be combined with other rebuild? Some customer has already verified triggering rebuild during some target reintegration.

Reply:
> discard version will be set as long as the target status is ST_UP (in reintegration), that may be combined with other rebuild?

I think it does not matter if it combined with other rebuild, as long as this discard from reintegration (UP), in which case we need discard the entry anyway.

Reply:
Yes, we need to discard some DTX entries, but just for old DTX entries, not for all.